File: Java/DivisibilityBy29.java
public class DivisibilityBy29 {

	public static void main(String[] args) {
		for(int i = 9999; i >= 1000; -- i)
			if(i % 29 == 0 && isTheDigitSum29(Integer.toString(i).toCharArray()))
				System.out.println(i);
	}
	
	private static boolean isTheDigitSum29(char[] array)
	{
		long sum = 0L;
		
		for(char c: array)
		{
			sum += c - 48;
			
			if(sum > 29)
				return false;
		}
		
		return sum == 29;
	}
}

File: DivisibilityBy29.py
    def isTheDigitSum29(number):
        sum = 0;
        length = len(number);
        
        for i in range(0, length, 1):
            sum += ord(number[i]) - 48;
            
            if(sum > 29):
                return False;
                
        return sum == 29;
        
    for i in range(9999, 1000, -1):
        if(i % 29 == 0 and isTheDigitSum29(str(i))):
            print(i);
